File: src/main/java/org/manapart/item_filters/EnderExtenderBlock.java
package org.manapart.item_filters;

import net.minecraft.block.*;
import net.minecraft.block.material.Material;
import net.minecraft.block.material.MaterialColor;
import net.minecraftforge.common.ToolType;

public class EnderExtenderBlock extends Block {

    public EnderExtenderBlock() {
        super(createProps());
    }

    private static Properties createProps() {
        Material padMat = new Material.Builder(MaterialColor.COLOR_BLUE).build();
        Properties props = Properties.of(padMat);
        props.requiresCorrectToolForDrops();
        props.harvestTool(ToolType.PICKAXE);
        props.sound(SoundType.METAL);
        props.strength(4);
        return props;
    }

}

File: src/main/java/org/manapart/endersort/EnderExtenderBlock.kt
package org.manapart.endersort

import net.minecraft.world.level.block.EndRodBlock
import net.minecraft.world.level.block.RodBlock
import net.minecraft.world.level.block.SoundType
import net.minecraft.world.level.block.state.BlockBehaviour
import net.minecraft.world.level.material.Material
import net.minecraft.world.level.material.MaterialColor


class EnderExtenderBlock : EndRodBlock(createProps())

private fun createProps(): BlockBehaviour.Properties {
    val padMat = Material.Builder(MaterialColor.COLOR_BLUE).build()
    val props = BlockBehaviour.Properties.of(padMat)
    props.requiresCorrectToolForDrops()
    props.sound(SoundType.METAL)
    props.strength(4f)
    return props
}
